### PR TITLE
optimize single-pass f32 parsing

### PR DESCRIPTION
**Pull request by liuq19**
## Summary
- add a dedicated single-pass `f32` parser in `sonic-number`
- use that parser from `Deserializer::deserialize_f32`
- lock in overflow handling with explicit negative tests returning `FloatMustBeFinite`
- strengthen fuzz coverage with a dedicated `f32` literal target, curated seeds, and CI fuzz-time normalization

## Background
The previous `f32` deserialization path first parsed the number as `f64` and then converted it back to `f32`. For tie-boundary inputs such as `17005001.000000000000130`, that can produce an off-by-one ULP result.

## Changes
This updates `f32` parsing to stay on a single pass, avoids the extra string reparse fallback, and preserves the expected `f32` rounding behavior for affected cases.

On top of the parser change, this PR also strengthens regression coverage around `f32` parsing:
- add direct regression tests for precision-sensitive literals and overflow-to-infinity cases such as `3.4028236e38` and `1e39`
- add a dedicated `fuzz_f32_literal` target with curated corpus seeds for rounding and overflow boundaries
- expand fuzz seed coverage for JSON value, string, path, deep nesting, and serde roundtrip paths
- normalize `FUZZ_TIME` in CI so values like `3m` are passed to libFuzzer as seconds instead of effectively running for only a few seconds

## Validation
- `cargo test test_deserialize_f32_single_pass_precision`
- `cargo test --manifest-path sonic-number/Cargo.toml test_parse_float32`
- `cargo test --manifest-path fuzz/Cargo.toml`

### CODE REVIEW

File: sonic-number/src/lib.rs [hunk 1]
```diff
@@ -922,4 +1204,23 @@ mod test {
             3.469446951536142e-26,
         );
     }
+
+    #[test]
+    fn test_parse_float32() {
+        test_parse_f32_ok("0", 0.0);
+        test_parse_f32_ok("-0", -0.0);
+        test_parse_f32_ok("1", 1.0);
+        test_parse_f32_ok("0.1", 0.1);
```
[the remark this answers is not in the document]
Comment (liuq19) [reply sonic-number/src/lib.rs:1229]: Added explicit negative coverage for the new finite-only behavior. `parse_float32` now has direct assertions that `"3.4028236e38"` and `"1e39"` return `Error::FloatMustBeFinite`, and there is also a serde-level regression test for `from_str::<f32>("1e39")` to lock the behavior in end-to-end.

**Review by shenyj3 — APPROVED** [review]
